src/intrinsics/ecma262/Symbol.js

```diff
@@ -28,11 +28,12 @@ export default function(realm: Realm): NativeFunctionValue {
     let descString;
     if (!description || description instanceof UndefinedValue) {
       descString = undefined;
+    } else if (description instanceof AbstractValue) {
+      return new SymbolValue(realm, description);
```

Comment:
I would change to:
```
descString = description;
```
so that you can leverage the single "return new SymbolValue()" below.